File: java-implementation/src/main/java/com/inventory/products/clothing/Dresses.java
package com.inventory.products.clothing;

// Represents a concrete subclass of Clothing
public class Dresses extends Clothing {
    private String type;

    // Constructs an instance with specified details with base class attributes included
    public Dresses(int id, String name, double price, int volume, String size, String material, String type, String supplier) {
        super(id, name, price, volume, size, material, supplier);
        this.type = type;
    }

    // Calculates the discount based on volume and month
    @Override
    public double calculateDiscount(int volume, int month) {
        if (month == 6 && volume > 3) {
            return price * volume * 0.50;
        }
        return 0;
    }

    // Provides a string representation the details
    @Override
    public String displayProductInfo() {
        return "ID: " + id + "\nName: " + name + "\nSize: " + size
               + "\nPrice: €" + price + "\nMaterial: " + material
               + "\nType: " + type + "\nVolume: " + volume + " units\nSupplier: " + supplier + "\n\n";
    }
}

File: cpp-implementation/src/products/Clothing/Dresses.cpp
#include "Product.ih"

// Updated constructor implementation for dresses
Dresses::Dresses(int id, string name, double price, int volume,
                 string size, string material, string type, string supplier)
    : Clothing(id, name, price, volume, size, material, supplier),
      type(type) {}

// Calculates the discount depending on the current month and the packaged quantity
double Dresses::calculateDiscount(int volume, int month) const {
    if (month == 6 && volume > 3) {
        return price * volume * 0.50;
    }
    return 0;
}

// Displays all the information
string Dresses::displayProductInfo() const {
    ostringstream info;
    info << "ID: " << id << "\nDress: " << name << "\nSupplier: " << getSupplier()
         << "\nSize: " << size << "\nMaterial: " << material
         << "\nType: " << type << "\nPrice: €" << price
         << "\nVolume: " << volume << " units\n\n";
    return info.str();
}
